sync/src/main/java/tech/pegasys/teku/sync/forward/multipeer/SyncController.java

```diff
@@ -123,15 +100,20 @@ private void onSyncComplete(final SyncResult result) {
         currentSync.map(Objects::toString).orElse("<unknown>"),
         result);
     // See if there's a new sync we should start (possibly switching to non-finalized sync)
-    currentSync = selectNewSyncTarget(currentSync.map(InProgressSync::getTargetChain));
+    currentSync = selectNewSyncTarget();
     if (!isSyncActive()) {
       currentSync = Optional.empty();
       notifySubscribers(false);
     }
   }
 
   public boolean isSyncActive() {
-    return currentSync.map(InProgressSync::isActive).orElse(false);
+    return isSyncActive(currentSync);
+  }
+
+  public boolean isSyncActive(final Optional<InProgressSync> sync) {
+    return sync.map(inProgressSync -> inProgressSync.isActive() && !inProgressSync.isSpeculative())
+        .orElse(false);
```

Comment:
Might be clearer to add a helper method to try to encapsulate this idea (active, not speculative):
```suggestion
    return sync.map(inProgressSync -> inProgressSync.isActivePrimarySync())
        .orElse(false);
```

Reply:
Done.